### PR TITLE
Implement MasterXPubDAO

### PR DESCRIPTION
**Pull request by Christewart**
This is the first step in addressing #3447

This PR also implements validating the master xpub stored in the database against the actual key manager on startup for `wallet`, `oracle`. This means you will get an exception on startup if they aren't the same.

### CODE REVIEW

File: db-commons/src/main/scala/org/bitcoins/db/models/MasterXPubDAO.scala [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import org.bitcoins.core.crypto.{ChainCode, ExtKeyPubVersion, ExtPublicKey}
+import org.bitcoins.core.number.{UInt32, UInt8}
+import org.bitcoins.crypto.ECPublicKey
+import org.bitcoins.db.{CRUD, DbAppConfig, SlickUtil}
+import scodec.bits.ByteVector
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/** The primary key type is the fingerprint [[ExtPublicKey.fingerprint]] */
```
Comment (Christewart) [inline db-commons/src/main/scala/org/bitcoins/db/models/MasterXPubDAO.scala]: old, needs to change

File: db-commons/src/main/scala/org/bitcoins/db/DbCommonsColumnMappers.scala [hunk 2]
```diff
@@ -5,11 +5,11 @@ import org.bitcoins.commons.jsonmodels.wallet.{
   WalletStateDescriptorType
 }
 import org.bitcoins.core.config.{BitcoinNetwork, BitcoinNetworks}
-import org.bitcoins.core.crypto._
+import org.bitcoins.core.crypto.{ExtKeyPubVersion, _}
```
Comment (benthecarman) [inline db-commons/src/main/scala/org/bitcoins/db/DbCommonsColumnMappers.scala:8]: nit: `_`

File: db-commons/src/main/scala/org/bitcoins/db/models/MasterXPubDAO.scala [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      if (xpub != xpubs.head) {
+        sys.error(
+          s"Keymanager xpub and stored xpub are different, stored=${xpubs.head}, keymanager=${xpub}")
+      }
+    }
+  }
+
+  def existsOneXpub(): Future[Boolean] = {
+    safeDatabase.run(table.exists.result)
+  }
```
Comment (benthecarman) [inline db-commons/src/main/scala/org/bitcoins/db/models/MasterXPubDAO.scala]: this only checks if the table exists, not if there is only 1 xpub

**Review by benthecarman — COMMENTED** [review]
otherwise lgtm

**Comment by Christewart** [comment]
oops, didn't mean to re-request review yet @benthecarman . Apologies

**Pull request merged**